opentech/apply/activity/messaging.py

```diff
@@ -82,6 +89,56 @@ def send_message(self, message, user, submission, **kwargs):
         )
 
 
+class SlackAdapter(AdapterBase):
+    adapter_type = "Slack"
+    always_send = True
+    messages = {
+        MESSAGES.NEW_SUBMISSION: 'A new submission has been submitted for {{submission.page.title}}: <{link}|{submission.title}>',
+        MESSAGES.UPDATE_LEAD: 'The lead of <{link}|{submission.title}> has been updated from {old.lead} to {submission.lead} by {user}',
+        MESSAGES.COMMENT: 'A new comment has been posted on <{link}|{submission.title}>',
+        MESSAGES.REVIEWERS_UPDATED: '{user} has updated the reviewers on <{link}|{submission.title}>',
+        MESSAGES.TRANSITION: '{user} has updated the status of <{link}|{submission.title}>: {old_phase.display_name} → {submission.phase}',
+        MESSAGES.DETERMINATION_OUTCOME: 'A determination for <{link}|{submission.title}> was sent by email: {submission.determination.get_outcome_display}',
+        MESSAGES.PROPOSAL_SUBMITTED: 'A proposal has been submitted for review: <{link}|{submission.title}>',
+        MESSAGES.INVITED_TO_PROPOSAL: '<{link}|{submission.title}> by {submission.user} has been invited to submit a proposal',
+        MESSAGES.NEW_REVIEW: '{user} has submitted a review for <{link}|{submission.title}>. Outcome: {review.outcome} Score: {review.score}'
+    }
+
+    def __init__(self):
+        super().__init__()
+        self.destination = settings.SLACK_DESTINATION_URL
+        self.target_room = settings.SLACK_DESTINATION_ROOM
+
+    def message(self, message_type,  **kwargs):
+        user = kwargs['user']
+        submission = kwargs['submission']
+        request = kwargs['request']
+        link = link_to(submission, request)
+
+        message = super().message(message_type, link=link, **kwargs)
+
+        if submission.lead.slack:
+            slack_target = self.slack_id(submission.lead)
+        else:
+            slack_target = ''
+
+        message = ' '.join([slack_target, message]).strip()
+        return message
+
+    def slack_id(self, user):
+        return f'<{user.slack}>'
+
+    def send_message(self, message, **kwargs):
+        if not self.destination and not self.target_room:
```

Comment:
Should this not return when either destination or target_room is not set?
i.e. `if not self.destination or not self.target_room:`